frontend/dockerfile/dockerfile2llb/convert.go

```diff
@@ -294,6 +294,15 @@ func Dockerfile2LLB(ctx context.Context, dt []byte, opt ConvertOpt) (*llb.State,
 								}
 							}
 						}
+						if !isScratch {
+							img.BuildInfo, _ = json.Marshal(exptypes.BuildInfo{
```

Comment:
I don't think there is a reason to add this to img here. just add property directly to dispatchState (without marshaling)